### ISSUE TITLE
Autocomplete feezes GUI

### ISSUE THREAD

**Issue by xmedeko** [HeidiSQL/HeidiSQL#10]
### Steps to reproduce

1. Connect to a MariaDB.
2. Run a long time query, e.g.:
```select sleep(120);```
3. Start typing a new query in the editor. At some point, the HeidiSQL tries to autocomplete the column names and the GUI freezes until the first query finishes.

## Expected behavior

Do not freeze the GUI. I suppose either one of:
* Use separate DB connection for fetching the DB metadata.
* Freeze the editor only, not the whole GUI, so as it is possible to cancel the query (ESC).
* Disable autocomplete and other operations requiring the metadata when a query is running. (This is the least convenient, since the query may finish soon.)

### Context

* HeidiSQL version: 9.4.0.5185
* Database system + version: MariaDB 10.0.31
* Operating system: Windows 10

**Comment by ansgarbecker** [comment]
Similar forum thread: https://www.heidisql.com/forum.php?t=23387

**Comment by xmedeko** [comment]
@jcguevara Yep, that's what I have meant by "Use separate DB connection for fetching the DB metadata."
